pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/scheduler/IScheduler.java

```diff
@@ -38,4 +39,12 @@
      */
     List<Assignment> schedule(List<Instance> unassignedFunctionInstances, List<Assignment> currentAssignments,
             Set<String> workers);
+
+    /**
+     *
+     * @param currentAssignments
+     * @param workers
+     * @return
+     */
+    List<Function.Assignment> rebalance(List<Function.Assignment> currentAssignments, Set<String> workers);
```

Comment:
Should we have a default implementation that just returns the current assignment?

Reply:
Sure we can do it for BC reasons.  It would return a empty list because there will be no change